platform.txt

```diff
@@ -129,3 +129,26 @@ tools.openocd.erase.pattern=
 tools.openocd.bootloader.params.verbose=-d2
 tools.openocd.bootloader.params.quiet=-d0
 tools.openocd.bootloader.pattern="{path}/{cmd}" {bootloader.verbose} -f interface/{program.protocol}.cfg -c "{program.setup_command}" -f target/{upload.target}.cfg -c "init; halt; nrf51 mass_erase; program {{{runtime.platform.path}/cores/nRF5/SDK/components/softdevice/{softdevice}/hex/{softdevice}_{upload.target}_{softdeviceversion}_softdevice.hex}} verify reset; shutdown;"
+
+# blackmagic probe upload
+tools.bmp_upload.cmd=arm-none-eabi-gdb
+tools.bmp_upload.path={runtime.tools.gcc-arm-none-eabi-5_2-2015q4.path}/bin/
```

Comment:
Could we use `{compiler.path}` here or another key, just to make it easier to upgrade the toolchain in the future?

Reply:
That's a direct copy from the openocd path in boards.txt. Ill investigate (because im not familiar enough with this work) and make changes if there is no issue

Reply:
@sandeepmistry - doesnt look like we have access to this as a global parameter : 
[Arduino-IDE-1.5-3rd-party-Hardware-specification](https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5-3rd-party-Hardware-specification).

It might be better to add a runtime define path to the gcc tools and openocd around line 20, like they do in 
[esp8266 platform.txt](https://github.com/esp8266/Arduino/blob/master/platform.txt)